### ISSUE TITLE
When using a non-default path for swagger-ui the js and css resources cannot be loadad

### ISSUE THREAD

**Issue by gaetjen** [micronaut-projects/micronaut-openapi#910]
This was introduced when I upgraded Micronaut from 3.7.4 to 3.8.0.

### Expected Behavior

When setting the following properties in `application.yml` the swagger-ui should work at `baseurl/customPath`
```yaml
micronaut:
  router:
    static-resources:
      swagger:
        paths: classpath:META-INF/swagger
        mapping: /swagger/**
      swagger-ui:
        paths: classpath:META-INF/swagger/views/swagger-ui
        mapping: /customPath/**
```

### Actual Behaviour

The `index.html`still uses the default paths to reference js and css resources.
Excerpt:
```html
<script src='/swagger-ui/res/swagger-ui-bundle.js'></script>
<script src='/swagger-ui/res/swagger-ui-standalone-preset.js'></script>
<link rel='stylesheet' type='text/css' href='/swagger-ui/res/swagger-ui.css' />
<link rel='stylesheet' type='text/css' href='/swagger-ui/res/flattop.css' />
```
As a result, the swagger-ui page is just blank.

### Steps To Reproduce

Build and run linked repo, open http://localhost:8080/customPath

### Environment Information

- Ubuntu version 22.04
- Jdk version: 17
- Language: Kotlin

### Example Application

https://github.com/gaetjen/openapi-bug

### Micronaut Version

3.8.0, 3.8.1

### Workaround

A possible, but not ideal workaround is to explicitly set the resources path back to the default in the application.yml:
```yaml
      swagger-ui-res:
        paths: classpath:META-INF/swagger/views/swagger-ui/res
        mapping: /swagger-ui/res/**
```
Or you can use the default location for the swagger-ui (i.e. http://baseUrl/swagger-ui)

**Comment by altro3** [comment]
@gaetjen This feature has been around for a long time. Look at the documentation, you need a property `swagger-ui.js.url`:
```
swagger-ui.js.url -   Override  path to use to find the js file. Path must contain a files named  swagger-ui.css, swagger-ui-bundle.js, swagger-ui-standalone-present.js.  The value can be a URL path like: "https://unpkg.com/swagger-ui-dist/".  Must contain trailing slash.,
```
For your case you need add properties like this:
```groovy
options.forkOptions.jvmArgs += "-Dmicronaut.openapi.views.spec=swagger-ui.enabled=true,swagger-ui.js.url=/customPath/res/"
```

**Comment by gaetjen** [comment]
Ah, I did try setting the `swagger-ui.js.url` in the `openapi.properties` file but it wasn't working. I think the reason was that you need to do a `gradle clean` first in order to get the changes to show up in the running application. I got it to work now on [this branch](https://github.com/gaetjen/openapi-bug/tree/set_js_url_property).

But also note, that using this fix does not work with Micronaut version 3.7 (tested on 3.7.4), I believe because of the issue described [here](https://github.com/micronaut-projects/micronaut-openapi/issues/823). See [this branch](https://github.com/gaetjen/openapi-bug/tree/mn3.7_broken).

So [the workaround](https://github.com/gaetjen/openapi-bug/tree/mn3.7_workaround) for version 3.7.4 involves setting a mapping for the swagger-ui resources, but does not include setting the `swagger-ui.js.url` property. And when you then [upgrade to Micronaut 3.8](https://github.com/gaetjen/openapi-bug/tree/workaround_for_3.7_breaks_on_3.8) it breaks.

So because we had the workaround for the other issue in our project I had some trouble understanding what the intended correct behavior and settings were supposed to be.
